driver/src/contract.rs

```diff
@@ -59,9 +59,8 @@ impl SnappContractImpl {
         let snapp_base: serde_json::Value = serde_json::from_str(&contents)?;
         let snapp_base_abi: String = snapp_base.get("abi").ok_or("No ABI for contract")?.to_string();
 
-        let snapp_address = hex::decode(&(env::var("SNAPP_CONTRACT_ADDRESS")?)[2..])?;
-        let address: Address = Address::from(&snapp_address[..]);
-        let contract = Contract::from_json(web3.eth(), address, snapp_base_abi.as_bytes())?;
+        let address: Address = (env::var("SNAPP_CONTRACT_ADDRESS")?)[2..22].parse()?;
+        let contract = Contract::from_json(web3.eth(), address, snapp_base_abi.as_bytes()).unwrap();
```

Comment:
I want to avoid the unwrap here, but then I get:

```
the trait bound `error::DriverError: std::convert::From<ethabi::errors::Error>` is not satisfied
```
I am implementing the trait:
```
impl From<ethabi::errors::Error> for DriverError {
    fn from(error: ethabi::errors::Error) -> Self {
        DriverError::new(error.description(), ErrorKind::AbiError)
    }
}
```
it keeps on telling me that the trait ethabi::errors is private.

Reply:
So you are trying to remove the `unwrap`. In this case we should expect to get a return value of type `Result<Contract<T>, ethabi::Error>`

Reply:
Yes, I can tell you whats wrong, here... Might have something to do with a [bug report](https://github.com/intellij-rust/intellij-rust/issues/3654) I filed a while back.

Notice that my IDE thinks the return type is `Contract<http>` is this expected?

![Screenshot 2019-06-17 at 9 40 19 AM](https://user-images.githubusercontent.com/11778116/59586834-16255780-90e4-11e9-8bed-bdbde23fddbb.png)

Reply:
Actually, the only problem I am experiencing when removing the unwrap is the mismatched types in the next line

```
error[E0308]: mismatched types
  --> src/contract.rs:64:32
   |
64 |         Ok(SnappContractImpl { contract, web3, event_loop })
   |                                ^^^^^^^^ expected struct `web3::contract::Contract`, found enum `std::result::Result`
```

Reply:
ethabi::error seems to be public here: https://docs.rs/ethabi/8.0.0/ethabi/struct.Error.html

Reply:
> o you are trying to remove the unwrap. In this case we should expect to get a return value of type Result<Contract<T>, ethabi::Error>

sorry, I wanted to replace unwrap with error propagation done with a ?

>ethabi::error seems to be public here: https://docs.rs/ethabi/8.0.0/ethabi/struct.Error.html
Nice, an update to a newer version helped :)